realm/realm-library/src/androidTest/java/io/realm/RealmQueryTests.java

```diff
@@ -1520,6 +1630,19 @@ public void endsWith_nullForNullableStrings() {
                 (String) null).findFirst().getFieldStringNotNull());
     }
 
+    // Querying nullable field with like - nulls do not match either '?' or '*'
+    @Test
+    public void like_nullForNullableStrings() {
+        TestHelper.populateTestRealmForNullTests(realm);
+        RealmResults<NullTypes> resultList = realm.where(NullTypes.class)
+                .like(NullTypes.FIELD_STRING_NULL, "*").findAll();
+        assertEquals(2, resultList.size());
```

Comment:
So null matches `*` but not `?`? It seems to be confused from the comments of this function.